pytorch_lightning/core/datamodule.py

```diff
@@ -366,3 +362,54 @@ def get_init_arguments_and_types(cls) -> List[Tuple[str, Tuple, Any]]:
             name_type_default.append((arg, arg_types, arg_default))
 
         return name_type_default
+
+    @classmethod
+    def from_datasets(
+        cls,
+        train_dataset: Optional[Dataset] = None,
```

Comment:
with #1959 merged we should allow for both dataset, list of dataset and dict of dataset
```suggestion
        train_dataset: Optional[Union[Dataset, List[Dataset], Dict[str, Dataset]] = None,
```